### PR TITLE
GDB-10224 Fixed the deployment of key vault diagnostic settings when monitoring is enabled

### PR DESCRIPTION
**Pull request by viktor-ribchev**
## Description

 Fixed the deployment of key vault diagnostic settings when monitoring is enabled

## Related Issues

GDB-10224

## Changes

Introduced new variable `create_key_vault_diagnostic_settings`

## Screenshots (if applicable)

N/A

## Checklist

- [X] I have tested these changes thoroughly.
- [X] My code follows the project's coding style.
- [X] I have added appropriate comments to my code, especially in complex areas.
- [X] All new and existing tests passed locally.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -208,8 +208,9 @@ module "monitoring" {
   ag_notifications_email_list = var.notification_recipients_email_list
 
   # Diagnostic settings
-  app_configuration_id = module.appconfig.app_configuration_id
-  key_vault_id         = var.tls_certificate_id != null ? null : module.vault[0].key_vault_id
+  app_configuration_id                 = module.appconfig.app_configuration_id
+  key_vault_id                         = var.tls_certificate_id != null ? null : module.vault[0].key_vault_id
+  create_key_vault_diagnostic_settings = var.tls_certificate_id == null ? "true" : "false"
```
Comment (mihailradkov) [inline main.tf:213]: Provide them as booleans, not as strings.

**Review by simeonzhekofff — APPROVED** [review]
LGTM

File: modules/monitoring/diagnostic_settings.tf [hunk 2]
```diff
@@ -13,7 +13,7 @@ resource "azurerm_monitor_diagnostic_setting" "application_config_diagnostic_set
 # Key Vault Audit log monitoring
 
 resource "azurerm_monitor_diagnostic_setting" "key_vault_diagnostic_settings" {
-  count = var.key_vault_id != null ? 1 : 0
+  count = var.create_key_vault_diagnostic_settings ? 1 : 0
```
Comment (mihailradkov) [inline modules/monitoring/diagnostic_settings.tf:16]: You could update the change log with this fix

**Review by mihailradkov — APPROVED** [review]

File: modules/monitoring/diagnostic_settings.tf (hunk 2 above)
Comment (viktor-ribchev) [reply modules/monitoring/diagnostic_settings.tf:16]: Updated

**Pull request merged**